src/Controller/Admin/ActivityController.php

```diff
@@ -83,11 +119,14 @@ public function newAction(Request $request): Response
      */
     public function showAction(Activity $activity): Response
     {
+        $this->denyAccessUnlessGranted('in_group', $activity->getAuthor());
```

Comment:
In fact this is the only solution, as the `$activity` variable is necessary and doesn't get assigned earlier than when the controller is called, since it's a route variable. Feel free to open up an issue if you have any ideas for how to improve this. Until then, I'm marking this conversation as resolved for now.

Reply:
This one, and the `null !== vs !is_null`, were meant to see how you thought about them. I assumed my comments made this clear, apparently this wasn't the case.

From your general reaction I get the feeling that you mostly agree with my feedback, I'll make an issue for the `is_null` and will keep this one in my backlog to think about.

Reply:
It was clear in this case, but my opinion is that there is that I can't think of a better solution than this, given the constraints with the Activity ;) (and with regards to the `null !== vs !is_null` as you call it, my opinion there is that I have no opinion)